### PR TITLE
Semroc change

### PR DESCRIPTION
**Pull request by thzero**
There is an issue with the Semroc file; OpenRocket can't handle the " character in the PartNumber as the PartNumber is stored in an attribute on an XML element.

Also, not sure, but the %Homepath%\openrocket\components doesn't work under the latest Win10.  I got it to work with %homepath%\appdata\romaing\openrocket\components

### CODE REVIEW

**Comment by dbcook** [comment]
Thanks for the Windows install info, I'm definitely going to incorporate that (and remove the previous since it nobody has confirmed it as good).

I'm puzzled by the XML change since " is not a character that must be escaped in an element value.  It would in an attribute value, but those are element values, not attribute values.  Everything works for me, there is no complaint on loading the file and I can select and insert the components fine.  Does it fail/complain for you?

image

File: README.md [hunk 1]
```diff
@@ -145,6 +145,14 @@ Run the following:
 
 ```bash
 git clone https://github.com/dbcook/openrocket-database.git
+mklink /D %HOMEPATH%\AppData\Roaming\OpenRocket\Components C:\openrocket-database\orc
```
Comment (dbcook) [inline README.md]: is this the Windows 10 version?  (with AppData\Roaming)

**Comment by dbcook** [comment]
After doing some research, it appears we should be able to use %APPDATA%, which should take care of the roaming vs non-roaming case automatically, i.e. use

mklink /D %APPDATA%\OpenRocket\Components C:\openrocket-database\orc

Can you confirm if this works?  It looks like a more general solution.

**Comment by thzero** [comment]
Yup.. fails for me.   Java8 with latest OR.   It fails when opening a saved
file. Error is an invalid xml.

So I checked.  Renane Ork file as zip.  Extract rocket.ork (why a small
text file needs to be zipped is beyond me).  Xml file is indeed invalid
because it had an attribute that looks like

Attrname="somenamendimginquote""

That seems to make it invalid when reading the xml file back into OR.

**Comment by dbcook** [comment]
Hmm, interesting!  The inscrutable workings of the .ork have become visible.  FWIW I think that should now be reported to OR as a bug; if they take something from an XML element value and internally throw it into an attribute value, they should encode it since the rules are not the same.

Meanwhile I'll accept this change as a workaround since I doubt they'll release a fix anytime soon :)

**Comment by dbcook** [comment]
Note to self - there may be similar in madcow.orc

**Comment by dbcook** [comment]
Hey can you check my rewrite of the windows install section that uses %APPDATA% (just pushed) and see if it works?

**Comment by thzero** [comment]
I'll check.

**Review by dbcook — APPROVED** [review]

**Comment by thzero** [comment]
I'm trying to get the dumb thing to open in intellij... Java is pita.

If I do.. there are some things like providing a nonarchived file extension.

But yeah.. I agree.

On Dec 11, 2017 10:46 AM, "Dave Cook" <notifications@github.com> wrote:

Hmm, interesting! The inscrutable workings of the .ork have become visible.
FWIW I think that should now be reported to OR as a bug; if they take
something from an XML element value and internally throw it into an
attribute value, they should encode it since the rules are not the same.

Meanwhile I'll accept this change as a workaround since I doubt they'll
release a fix anytime soon :)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/dbcook/openrocket-database/pull/2#issuecomment-350782952>,
or mute the thread

**Comment by thzero** [comment]
Thought I had checked that too.

**Comment by dbcook** [comment]
Madcow has seriously wacko part numbering, compounded by website code bugs

**Pull request merged**